### DISCUSSION TITLE
VisionOS example playback fails when receiving type 2 RTMP chunk

### DISCUSSION THREAD

**Discussion by lbcheng888** [HaishinKit/HaishinKit.swift#1383] [Q&A]
1.Build and run macOS Example
2.run [mediamtx](https://github.com/bluenviron/mediamtx) live streaming server locally
3.publish a RTMP stream and playback
4.playing stops when appending type 2 chunk.
It was expected to play stream steady because FFmpeg play it well

**Comment by onebuckgames** [comment]
Looks like it fails on "publish" step.

**Comment by shogo4405** [comment]
If it's the same stream instance, you can only publish or play.

**Reply by lbcheng888** [reply]
publish and playback , it plays untils appending some type 2 chunk

**Reply by shogo4405** [reply]
It may not work properly depending on server-side quirks and broadcaster settings.I have tested it with Nginx, Wowza, AWS, and Cloudflare. Without logs or similar, it's hard to diagnose.

Please provide the following information:
[Server]
- name
- url
- Version

[Ingest]
- Software
    - name: for example OBS
    - Url
    - version
- AudioSettings:
    - mono or stereo
    - Codec name
- VideoSettings:
    - Codec type: (H264 or HEVC or AV1)
    - Codec type: (x264, VideoToolbox(Software))

**Reply by lbcheng888** [reply]
[Server]
name: [mediamtx](https://github.com/bluenviron/mediamtx)
version: lastest
[Ingest]
Build and run HaishinKit macOS example, rtmp://127.0.0.1/live, port: 1935 by default
[Info] [com.haishinkit.HaishinKit] [IOAudioResampler.swift:239] setUp(_:) > inputFormat:<AVAudioFormat 0x600000bb5d60:  1 ch,  48000 Hz, Float32>,outputFormat:<AVAudioFormat 0x600000bb5e00:  1 ch,  48000 Hz, Float32>
2024-05-03 21:24:46.465 [Info] [com.haishinkit.HaishinKit] [AudioCodec.swift:149] makeAudioConverter() > inputFormat:<AVAudioFormat 0x600000bb5e00:  1 ch,  48000 Hz, Float32>,outputFormat:<AVAudioFormat 0x600000bb6080:  1 ch,  48000 Hz, aac  (0x00000002) 0 bits/channel, 0 bytes/packet, 1024 frames/packet, 0 bytes/frame>
Video codec seems to be H264

**Reply by shogo4405** [reply]
What's did you mean publish and playback?  What specific actions are you performing?

**Reply by shogo4405** [reply]
Did you mean this step?
```
1. Launch HaishinKit.app
2. Open “Ingest Test” menu
3. Publish 
4. Open "Playback Test” menu
5. Playback
```

It's not possible.

**Reply by lbcheng888** [reply]
The step is right, is it because I commented out videoInputFormat check for makeMetaData as a referrence of HPRTMP library. 
```swift
    /// Creates flv metadata for a stream.
    open func makeMetaData() -> ASObject {
        var metadata: [String: Any] = [:]
        // if videoInputFormat != nil {
            metadata["width"] = videoSettings.videoSize.width
            metadata["height"] = videoSettings.videoSize.height
            #if os(iOS) || os(macOS) || os(tvOS)
            metadata["framerate"] = frameRate
            #endif
            switch videoSettings.format {
            case .h264:
                metadata["videocodecid"] = FLVVideoCodec.avc.rawValue
            case .hevc:
                metadata["videocodecid"] = FLVVideoFourCC.hevc.rawValue
            }
            metadata["videodatarate"] = videoSettings.bitRate / 1000
        // }
        if audioInputFormat != nil {
            metadata["audiocodecid"] = FLVAudioCodec.aac.rawValue
            metadata["audiodatarate"] = audioSettings.bitRate / 1000
            if let outputFormat = mixer.audioIO.outputFormat {
                metadata["audiosamplerate"] = outputFormat.sampleRate
            }
        }
        return metadata
    }
```
And for clarification, the reason why I choose to reproduce it as above is that I found it results in the similar issue with visionOS Example when playing the stream without building and running another instance.
The issue is like

> 2024-06-03 20:38:49.694 [Info] [com.haishinkit.HaishinKit.visionOSApp] [ViewModel.swift:21] startPlaying() > playing:rtmp://127.0.0.1
2024-06-03 20:38:49.729 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:168] stateDidChange(to:) > Connection is ready.
2024-06-03 20:38:49.730 [Info] [com.haishinkit.HaishinKit] [RTMPNWSocket.swift:159] viabilityDidChange(to:) > Connection viability changed to true
NetConnection.Connect.Success
NetStream.Play.Reset
NetStream.Play.Start
2024-06-03 20:38:49.759 [Warn] [com.haishinkit.HaishinKit] [RTMPChunk.swift:251] append(_:message:) > ByteArray{data: 47 bytes,position: 1}
2024-06-03 20:38:49.759 [Error] [com.haishinkit.HaishinKit] [RTMPChunk.swift:153] data > 166 bytes
